core/src/main/scala/akka/persistence/postgres/journal/dao/ByteArrayJournalDao.scala

```diff
@@ -92,35 +91,24 @@ trait BaseByteArrayJournalDao extends JournalDaoWithUpdates with BaseJournalDaoW
   /**
    * @see [[akka.persistence.journal.AsyncWriteJournal.asyncWriteMessages(messages)]]
    */
-  def asyncWriteMessages(messages: Seq[AtomicWrite]): Future[Seq[Try[Unit]]] = {
-    // If serialization fails for some AtomicWrites, the other AtomicWrites may still be written
-    ensureTagForMessagesExists(messages).map(_ => serializer.serialize(messages)).flatMap { serializedTries =>
-      val rowsToWrite: Seq[JournalRow] = for {
-        serializeTry: Try[Seq[JournalRow]] <- serializedTries
-        row: JournalRow <- serializeTry.getOrElse(Seq.empty)
-      } yield row
-
-      def resultWhenWriteComplete =
-        if (serializedTries.forall(_.isSuccess)) Nil else serializedTries.map(_.map(_ => ()))
-
-      queueWriteJournalRows(rowsToWrite).map(_ => resultWhenWriteComplete)
-    }
-  }
-
-  private def ensureTagForMessagesExists(messages: Seq[AtomicWrite]): Future[Unit] = {
-    Future.sequence(extractTags(messages).map(eventTagConverter.getIdByName)).map(_ => ())
-  }
+  def asyncWriteMessages(messages: Seq[AtomicWrite]): Future[Seq[Try[Unit]]] =
+    Future
+      .sequence {
+        serializer
+          .serialize(messages)
+          // If serialization fails for some AtomicWrites, the other AtomicWrites may still be written
+          .map(_.map(Success(_)).recover {
+            case ex =>
+              Failure(ex)
+          })
```

Comment:
Maybe put it in serialize method?

Reply:
I'm not sure. Wrapping failed serialization in a `scala.util.Failure` is used there in order to allow JournalDao persist the rest of events that have been successfuly encoded. Change of DAO's behaviour should not affect the serializer component.